custom_components/myskoda/coordinator.py

```diff
@@ -168,27 +169,27 @@ async def _on_charging_event(self, event: EventCharging):
         else:
             status = vehicle.charging.status
 
-            if data.charged_range is not None:
+            if isinstance(event.event, ServiceEventChangeSoc):
                 status.battery.remaining_cruising_range_in_meters = (
                     data.charged_range * 1000
                 )
-            if data.soc is not None:
                 status.battery.state_of_charge_in_percent = data.soc
-            if data.time_to_finish is not None:
-                status.remaining_time_to_fully_charged_in_minutes = data.time_to_finish
-            if data.state is not None:
+                if data.time_to_finish is not None:
+                    status.remaining_time_to_fully_charged_in_minutes = (
+                        data.time_to_finish
+                    )
```

Comment:
Please format this using `match` and `case`, check https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching

We are python3.10+ (since HA is 3.12 and going to 3.13)

You could try to match the instance type

Reply:
Done